test/extensions/clusters/aggregate/cluster_integration_test.cc

```diff
@@ -0,0 +1,247 @@
+
+#include "absl/synchronization/notification.h"
+#include "gtest/gtest.h"
+
+using testing::AssertionResult;
+
+namespace Envoy {
+namespace {
+
+const char ClusterName1[] = "cluster_1";
+const char ClusterName2[] = "cluster_2";
+const int UpstreamIndex1 = 2;
+const int UpstreamIndex2 = 3;
+
+const std::string& config() {
+  CONSTRUCT_ON_FIRST_USE(std::string, R"EOF(
+admin:
+  access_log_path: /dev/null
+  address:
+    socket_address:
+      address: 127.0.0.1
+      port_value: 0
+dynamic_resources:
+  cds_config:
+    api_config_source:
+      api_type: GRPC
+      grpc_services:
+        envoy_grpc:
+          cluster_name: my_cds_cluster
+      set_node_on_first_message_only: true
+static_resources:
+  clusters:
+  - name: my_cds_cluster
+    http2_protocol_options: {}
+    hosts:
+      socket_address:
+        address: 127.0.0.1
+        port_value: 0
+  - name: aggregate_cluster
+    connect_timeout: 0.25s
+    lb_policy: CLUSTER_PROVIDED
+    cluster_type:
+      name: envoy.clusters.aggregate
+      typed_config:
+        "@type": type.googleapis.com/envoy.config.cluster.aggregate.ClusterConfig
+        clusters:
+        - cluster_1
+        - cluster_2
+  listeners:
+    name: http
+    address:
+      socket_address:
+        address: 127.0.0.1
+        port_value: 0
+    filter_chains:
+      filters:
+        name: envoy.http_connection_manager
+        config:
+          stat_prefix: config_test
+          http_filters:
+            name: envoy.router
+          codec_type: HTTP1
+          route_config:
+            name: route_config_0
+            validate_clusters: false
+            virtual_hosts:
+              name: integration
+              routes:
+              - route:
+                  cluster: cluster_1
+                match:
+                  prefix: "/cluster1"
+              - route:
+                  cluster: cluster_2
+                match:
+                  prefix: "/cluster2"
+              - route:
+                  cluster: aggregate_cluster
+                match:
+                  prefix: "/aggregatecluster"
+              domains: "*"
+)EOF");
+}
+
+class AggregateIntegrationTest : public testing::TestWithParam<Network::Address::IpVersion>,
+                                 public HttpIntegrationTest {
+public:
+  AggregateIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, GetParam(), config()) {
+    use_lds_ = false;
+  }
+
+  void TearDown() override {
+    cleanUpXdsConnection();
+    test_server_.reset();
+    fake_upstreams_.clear();
+  }
+
+  void initialize() override {
+    use_lds_ = false;
+    setUpstreamCount(2);                                  // the CDS cluster
+    setUpstreamProtocol(FakeHttpConnection::Type::HTTP2); // CDS uses gRPC uses HTTP2.
+
+    defer_listener_finalization_ = true;
+    HttpIntegrationTest::initialize();
+
+    fake_upstreams_.emplace_back(new FakeUpstream(0, FakeHttpConnection::Type::HTTP1, version_,
```

Comment:
can you just do `setUpstreamCount(4)` above instead of manually creating these two upstreams ?

Reply:
I cannot do `setUpstreamCount(4)` here because CDS cluster and other clusters use different upstream protocols. I'd like to keep it as is.

Reply:
Yeah i'm fine with that, was just curious about the motivation